util/update_pcluster_configs.py

```diff
@@ -33,103 +34,47 @@
-            current_object = self.download(s3_bucket, s3_key)
-            logging.info("Current version: %s", json.dumps(current_object))
-            reverting_object = self.download(s3_bucket, s3_key, version_id)
-            self.upload(s3_bucket, s3_key, reverting_object, dryrun=dryrun)
+    :param old_doc: old object to diff
+    :param new_doc: new object to diff
+    :return: Raises an exception / logs diffs. Nothing returned.
+    """
+    try:
+        import dictdiffer
+
+        logging.info("Found following diffs: %s", list(dictdiffer.diff(old_doc, new_doc)))
+    except ImportError:
+        logging.warning("Install dictdiffer if you want to display diffs: pip install dictdiffer")
+
+    logging.info("Found the following new root keys: %s", new_doc.keys() - old_doc.keys())
+    logging.info("Checking that the new configuration file includes the old entries.")
+    _assert_document_is_included(old_doc, new_doc)
+
+
+def _assert_document_is_included(doc_to_be_included, doc):
+    for k, v in doc_to_be_included.items():
+        if k not in doc:
+            raise Exception(f"Key {k} not found in new doc")
+        if isinstance(v, collections.Mapping):
+            _assert_document_is_included(v, doc[k])
+        elif isinstance(v, list):
+            if not all(elem in doc[k] for elem in v):
+                raise Exception(f"Old list {v} is not included in new list {doc[k]}")
         else:
-            logging.info("Current version is already the requested one: %s", version_id)
-
-    @staticmethod
-    def validate_document(old_doc, new_doc):
-        try:
-            import dictdiffer
-
-            logging.info("Found following diffs: %s", list(dictdiffer.diff(old_doc, new_doc)))
-        except ImportError:
-            logging.warning("Install dictdiffer if you want to display diffs: pip install dictdiffer")
-
-        logging.info("Found the following new root keys: %s", new_doc.keys() - old_doc.keys())
-        logging.info("Checking that the new configuration file includes the old entries.")
-        S3JsonDocumentManager._assert_document_is_included(old_doc, new_doc)
-
-    @staticmethod
-    def _assert_document_is_included(doc_to_be_included, doc):
-        for k, v in doc_to_be_included.items():
-            if k not in doc:
-                raise Exception(f"Key {k} not found in new doc")
-            if isinstance(v, collections.Mapping):
-                S3JsonDocumentManager._assert_document_is_included(v, doc[k])
-            elif isinstance(v, list):
-                if not all(elem in doc[k] for elem in v):
-                    raise Exception(f"Old list {v} is not included in new list {doc[k]}")
-            else:
-                if v != doc[k]:
-                    raise Exception(f"Old value {v} does not match new value {doc[k]}")
+            if v != doc[k]:
+                raise Exception(f"Old value {v} does not match new value {doc[k]}")
 
 
-class ConfigGenerator(ABC):
+class _ConfigGenerator(ABC):
```

Comment:
is this really necessary? I mean the underscore for Classes? never saw that

Reply:
needs docstring without it, according to flake8